packages/apollo-federation/src/composition/validate/postComposition/__tests__/keyFieldsSelectInvalidType.test.ts

```diff
@@ -109,12 +109,45 @@ describe('keyFieldsSelectInvalidType', () => {
 
     const warnings = validateKeyFieldsSelectInvalidType(schema);
     expect(warnings).toMatchInlineSnapshot(`
-      Array [
-        Object {
-          "code": "KEY_FIELDS_SELECT_INVALID_TYPE",
-          "message": "[serviceA] Product -> A @key selects Product.price, which is a union type. Keys cannot select union types.",
-        },
-      ]
-    `);
+            Array [
+              Object {
+                "code": "KEY_FIELDS_SELECT_INVALID_TYPE",
+                "message": "[serviceA] Product -> A @key selects Product.price, which is a union type. Keys cannot select union types.",
+              },
+            ]
+        `);
+  });
+
+  it('warns if @key references fields of a list type', () => {
+    const serviceA = {
+      typeDefs: gql`
+        type Product @key(fields: "price") {
+          sku: String!
+          price: [Numeric]!
+        }
+
+        type Numeric {
+          euro: Int
+          usd: Int
+        }
+      `,
+      name: 'serviceA',
+    };
+
+    const serviceB = {
+      typeDefs: gql`
+        extend type Product {
+          sku: String! @external
+          name: String!
+        }
+      `,
+      name: 'serviceB',
+    };
+
+    const { schema, errors } = composeServices([serviceA, serviceB]);
+    expect(errors).toHaveLength(0);
+
+    const warnings = validateKeyFieldsSelectInvalidType(schema);
+    expect(warnings).toMatchInlineSnapshot(`Array []`);
```

Comment:
If this is supposed to be a warning, shouldn't the warning Array have something in it?

Reply:
Yep! That's what this PR is supposed to highlight. I'll add some more to the description